CONTRIBUTING.md

```diff
@@ -1,57 +1,77 @@
 # Contributing to Avalonia
 
-## Before You Start
+PRs are always welcomed from everyone. Following this guide will help us get your PR reviewed and merged as quickly as possible.
 
-Drop into our [telegram group](https://t.me/Avalonia) or [gitter chat room](https://gitter.im/AvaloniaUI/Avalonia) and let us know what you're thinking of doing. We might be able to give you guidance or let you know if someone else is already working on the feature.
+For this guide we're going to split PRs into two types: bug fixes and features; the requirements for each are slightly different.
 
-## Style
+## Bug Fixes
 
-The codebase uses [.net core](https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md) coding style.
+A bug fix will ideally be accompanied by tests. There are a few types of tests:
 
-Try to keep lines of code around 100 characters in length or less, though this is not a hard limit.
-If you're a few characters over then don't worry too much.
+- Unit tests are for issues that aren't related to platform features. These tests are located in the `tests` directly, categorised by the assembly which they're testing.
+- Integration tests are for issues that are related to platform features (for example fixing a bug with Window resizing). These tests are located in the `tests/Avalonia.IntegrationTests.Appium` directory. Integration tests should be run on Windows and macOS. See the readme in that directory for more information
+- Render tests are for issues with rendering. These tests are located in `tests/Avalonia.RenderTests` with separate project files for Skia and Direct2D. The Direct2D backend is currently mostly unmaintained so render tests that just run on Skia are acceptable
 
-**DO NOT USE #REGIONS** full stop.
+It's not always feasible to accompany a bug fix with a test, but doing so will speed up the review process.
 
-## Pull requests
+The commits in a bug fix PR **should follow this pattern**:
 
-A single pull request should be submitted for each change. If you're making more than one change,
-please submit separate pull requests for each change for easy review. Rebase your changes to make
-sense, so a history that looks like:
+- A commit with a failing unit test; followed by
+- A commit that fixes the issues
 
-* Add class A
-* Feature A didn't set Foo when Bar was set
-* Fix spacing
-* Add class B
-* Sort using statements
+In this way the reviewer can check out the commit with the failing test and confirm the problem. One this is confirmed, they can confirm the fix.
 
-Should be rebased to read:
+## Features
 
-* Add class A
-* Add class B
+**Features should be discussed with the core team before opening a PR.** Please open an issue to discuss the feature before starting work, to ensure that the core team are onboard.
```

Comment:
I agree with this change, it's just that the team is often very busy and therefore doesn't respond to discussions, I wouldn't want this to lead to fewer contributions in the long run.

Reply:
True and not true. It also leads to frustration if a PR doesn't get merged in the end. So hard to get a good measure here.

Reply:
Yes, this is unfortunate, but I prefer contributors not to waste days or weeks of time on something which isn't going to be accepted anyway. Of course no-one is stopping people opening undiscussed PRs, but with this guidance maybe they'll know what they're getting into.